src/Orleans.Core/Core/ClientBuilderExtensions.cs

```diff
@@ -90,6 +93,41 @@ public static IClientBuilder ConfigureLogging(this IClientBuilder builder, Actio
             return builder.ConfigureServices(collection => collection.AddLogging(loggingBuilder => configureLogging(loggingBuilder)));
         }
 
+        /// <summary>
+        /// Configures the client to connect to a silo on the localhost.
+        /// </summary>
+        /// <param name="gatewayPort">The local silo's gateway port.</param>
+        public static IClientBuilder UseLocalhostClustering(this IClientBuilder builder, int gatewayPort = 30000, string clusterId = null)
```

Comment:
why isn't clusterId to set to `ClusterOptions .DevelopmentClusterId` so we can save those `IsNullOrWhiteSpace` dance?

Reply:
generally speaking, the optional param should be set to its default value, instead of null and not being used later, right?

Reply:
Good point.
Yes, optional param should be the default value. The exception is for values which cannot be represented as compile-time constants, but that doesn't apply here.